framework/src/main/java/org/checkerframework/common/value/util/Range.java

```diff
@@ -1180,4 +1181,28 @@ private boolean isWithinHalfLong() {
     public boolean isWithinInteger() {
         return isWithin(Integer.MIN_VALUE, Integer.MAX_VALUE);
     }
+
+    /**
+     * Returns a Range representing all possible values for the given primitive type.
+     *
+     * @param typeKind one of INT, SHORT, BYTE, CHAR, or LONG
+     * @return the range for the given primitive type
+     */
+    public static Range create(TypeKind typeKind) {
```

Comment:
Could you move this method to the other `create` methods in this class?

Reply:
I am not trying to defy you  but this thing about moving function bodies into an order in the source file affects negatively way one traverses the history of a function (see blame.)
Such move makes it unnecessarily hard to deal with merging large independent concurrent features in parallel as simple side by side changes in time.
I submit to your consideration that 
1.  when we look at code via IDEs, the location of the function body is irrelevant as we can navigate with ease via outlines in any order desired.
2. when we have to look at large file differences here in github having only the minimum amount of change to deal with is better.
I submit that in general, adding methods at the bottom of large files and or renaming them in place has no negative effect on deltas and facilitates merging.

Reply:
Regarding the git history, I believe you made two arguments:
 * File organization does not matter.  I respectfully but strongly disagree.  I feel that it is much more important that the code in the file be logically organized, than that the git history be minimal.  I look at the code around a method more often than I look at its git history.  Even within an IDE, the context of a function matters.  It helps in understanding the code's design and how it fits together.
 * Moving the function would increase the size of the git history.  It would not.  Since this is new code, the git history won't be any bigger or smaller if the code is inserted into one location rather than another.

Regarding the difficulty of making multiple changes:  I agree with you, but that is a temporary problem that does not justify a messy organization of the file (which will hamper programmers long into the future).  What other pull requests depend on this one?  Is there anything that I can do to help with that merge?

Reply:
None of those bullets as you stated them.

I wrote and re-wrote this reply many times.  Long, short, middle ground. Writing lacks tone and is very much subject to misinterpretation among strangers.